python/cuml/test/dask/test_random_forest.py

```diff
@@ -334,14 +335,67 @@ def test_rf_regression_dask_cpu(partitions_per_worker, cluster):
         X_train_df, y_train_df = dask_utils.persist_across_workers(
             c, [X_train_df, y_train_df], workers=workers)
 
-        cu_rf_mg = cuRFR_mg(**cu_rf_params)
-        cu_rf_mg.fit(X_train_df, y_train_df)
+        cuml_mod = cuRFR_mg(**cu_rf_params)
+        cuml_mod.fit(X_train_df, y_train_df)
 
-        cu_rf_mg_predict = cu_rf_mg.predict(X_test, predict_model='CPU')
+        cuml_mod_predict = cuml_mod.predict(X_test, predict_model='CPU')
 
-        acc_score = r2_score(cu_rf_mg_predict, y_test)
+        acc_score = r2_score(cuml_mod_predict, y_test)
 
         assert acc_score >= 0.67
 
     finally:
         c.close()
+
+
+@pytest.mark.parametrize('partitions_per_worker', [1])
+@pytest.mark.parametrize('output_class', [True, False])
+def test_rf_classification_dask_fil_predict_proba(partitions_per_worker,
+                                                  cluster,
+                                                  output_class):
+
+    # Use CUDA_VISIBLE_DEVICES to control the number of workers
+    c = Client(threads_per_worker=1, n_workers=1)
+
+    try:
+
+        X, y = make_classification(n_samples=10000, n_features=30,
+                                   n_clusters_per_class=1, n_informative=20,
+                                   random_state=123, n_classes=2)
+
+        X = X.astype(np.float32)
+        y = y.astype(np.int32)
+
+        X_train, X_test, y_train, y_test = \
+            train_test_split(X, y, test_size=1000)
+
+        cu_rf_params = {'n_bins': 16, 'n_streams': 1,
+                        'n_estimators': 40, 'max_depth': 16
+                        }
+
+        X_train_df, y_train_df = _prep_training_data(c, X_train, y_train,
+                                                     partitions_per_worker)
+        X_test_df, _ = _prep_training_data(c, X_test, y_test,
+                                           partitions_per_worker)
+        cu_rf_mg = cuRFC_mg(**cu_rf_params)
+        cu_rf_mg.fit(X_train_df, y_train_df)
+
+        fil_preds_proba = cu_rf_mg.predict_proba(X_test_df).compute()
+        fil_preds_proba = cp.asnumpy(fil_preds_proba.to_gpu_matrix())
+        y_proba = np.zeros(np.shape(fil_preds_proba))
+        y_proba[:, 1] = y_test
+        y_proba[:, 0] = 1.0 - y_test
+        fil_mse = mean_squared_error(y_proba, fil_preds_proba)
+        sk_model = skrfc(n_estimators=40,
+                         max_depth=16,
+                         random_state=10)
+        sk_model.fit(X_train, y_train)
+        sk_preds_proba = sk_model.predict_proba(X_test)
+        sk_mse = mean_squared_error(y_proba, sk_preds_proba)
+
+        # The threshold is required as the test would intermitently
+        # fail with a max difference of 0.003 between the two mse values
+        assert fil_mse <= sk_mse + 0.003
```

Comment:
Testing MSE is one heavyweight option to compare to sklearn... what about a more targeted unit test that directly tests expected output probs?
 e.g. create a small synthetic dataset where X and y are uncorrelated and 75% of outcomes are class==1 and use a decent number of trees. Predict_proba should return roughly 75% for all new values right? then you could use a very small dataset and be confident we're returning probs.